2006-11-20  Jakub Jelinek  <jakub@redhat.com>

	* linux/ia64/syscallent.h: Adjust for SYS_socket_subcall
	bump, add #if to make sure this isn't forgotten again.

2006-08-21  Roland McGrath  <roland@redhat.com>

	* linux/syscall.h (SYS_socket_subcall): Bump to 400.
	* linux/syscallent.h: Update table.
	* linux/m68k/syscallent.h: Likewise.
	* linux/s390/syscallent.h: Likewise.
	* linux/s390x/syscallent.h: Likewise.
	* linux/powerpc/syscallent.h: Likewise.
	* linux/sh/syscallent.h: Likewise.
	* linux/sh64/syscallent.h: Likewise.
	Fixes RH#202620.

Index: strace/linux/syscall.h
diff -u strace/linux/syscall.h:1.46 strace/linux/syscall.h:1.47
--- strace/linux/syscall.h:1.46	Thu Jan 12 14:34:50 2006
+++ strace/linux/syscall.h	Mon Aug 21 23:06:12 2006
@@ -25,7 +25,7 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *
- *	$Id: syscall.h,v 1.46 2006/01/12 22:34:50 roland Exp $
+ *	$Id: syscall.h,v 1.47 2006/08/22 06:06:12 roland Exp $
  */
 
 #include "dummy.h"
@@ -183,7 +183,7 @@
 #  undef SYS_sendmsg
 #  undef SYS_recvmsg
 # endif /* IA64 */
-#  define SYS_socket_subcall	300
+#  define SYS_socket_subcall	400
 #define SYS_sub_socket		(SYS_socket_subcall + 1)
 #define SYS_sub_bind		(SYS_socket_subcall + 2)
 #define SYS_sub_connect		(SYS_socket_subcall + 3)
Index: strace/linux/syscallent.h
diff -u strace/linux/syscallent.h:1.39 strace/linux/syscallent.h:1.40
--- strace/linux/syscallent.h:1.39	Thu Jan 12 14:34:50 2006
+++ strace/linux/syscallent.h	Mon Aug 21 23:06:12 2006
@@ -25,7 +25,7 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *
- *	$Id: syscallent.h,v 1.39 2006/01/12 22:34:50 roland Exp $
+ *	$Id: syscallent.h,v 1.40 2006/08/22 06:06:12 roland Exp $
  */
 
 	{ 0,	0,	sys_restart_syscall,	"restart_syscall" }, /* 0 */
@@ -330,54 +330,154 @@
 	{ 5,	0,	printargs,		"SYS_297"	}, /* 297 */
 	{ 5,	0,	printargs,		"SYS_298"	}, /* 298 */
 	{ 5,	0,	printargs,		"SYS_299"	}, /* 299 */
+	{ 5,	0,	printargs,		"SYS_300"	}, /* 300 */
+	{ 5,	0,	printargs,		"SYS_301"	}, /* 301 */
+	{ 5,	0,	printargs,		"SYS_302"	}, /* 302 */
+	{ 5,	0,	printargs,		"SYS_303"	}, /* 303 */
+	{ 5,	0,	printargs,		"SYS_304"	}, /* 304 */
+	{ 5,	0,	printargs,		"SYS_305"	}, /* 305 */
+	{ 5,	0,	printargs,		"SYS_306"	}, /* 306 */
+	{ 5,	0,	printargs,		"SYS_307"	}, /* 307 */
+	{ 5,	0,	printargs,		"SYS_308"	}, /* 308 */
+	{ 5,	0,	printargs,		"SYS_309"	}, /* 309 */
+	{ 5,	0,	printargs,		"SYS_310"	}, /* 310 */
+	{ 5,	0,	printargs,		"SYS_311"	}, /* 311 */
+	{ 5,	0,	printargs,		"SYS_312"	}, /* 312 */
+	{ 5,	0,	printargs,		"SYS_313"	}, /* 313 */
+	{ 5,	0,	printargs,		"SYS_314"	}, /* 314 */
+	{ 5,	0,	printargs,		"SYS_315"	}, /* 315 */
+	{ 5,	0,	printargs,		"SYS_316"	}, /* 316 */
+	{ 5,	0,	printargs,		"SYS_317"	}, /* 317 */
+	{ 5,	0,	printargs,		"SYS_318"	}, /* 318 */
+	{ 5,	0,	printargs,		"SYS_319"	}, /* 319 */
+	{ 5,	0,	printargs,		"SYS_320"	}, /* 320 */
+	{ 5,	0,	printargs,		"SYS_321"	}, /* 321 */
+	{ 5,	0,	printargs,		"SYS_322"	}, /* 322 */
+	{ 5,	0,	printargs,		"SYS_323"	}, /* 323 */
+	{ 5,	0,	printargs,		"SYS_324"	}, /* 324 */
+	{ 5,	0,	printargs,		"SYS_325"	}, /* 325 */
+	{ 5,	0,	printargs,		"SYS_326"	}, /* 326 */
+	{ 5,	0,	printargs,		"SYS_327"	}, /* 327 */
+	{ 5,	0,	printargs,		"SYS_328"	}, /* 328 */
+	{ 5,	0,	printargs,		"SYS_329"	}, /* 329 */
+	{ 5,	0,	printargs,		"SYS_330"	}, /* 330 */
+	{ 5,	0,	printargs,		"SYS_331"	}, /* 331 */
+	{ 5,	0,	printargs,		"SYS_332"	}, /* 332 */
+	{ 5,	0,	printargs,		"SYS_333"	}, /* 333 */
+	{ 5,	0,	printargs,		"SYS_334"	}, /* 334 */
+	{ 5,	0,	printargs,		"SYS_335"	}, /* 335 */
+	{ 5,	0,	printargs,		"SYS_336"	}, /* 336 */
+	{ 5,	0,	printargs,		"SYS_337"	}, /* 337 */
+	{ 5,	0,	printargs,		"SYS_338"	}, /* 338 */
+	{ 5,	0,	printargs,		"SYS_339"	}, /* 339 */
+	{ 5,	0,	printargs,		"SYS_340"	}, /* 340 */
+	{ 5,	0,	printargs,		"SYS_341"	}, /* 341 */
+	{ 5,	0,	printargs,		"SYS_342"	}, /* 342 */
+	{ 5,	0,	printargs,		"SYS_343"	}, /* 343 */
+	{ 5,	0,	printargs,		"SYS_344"	}, /* 344 */
+	{ 5,	0,	printargs,		"SYS_345"	}, /* 345 */
+	{ 5,	0,	printargs,		"SYS_346"	}, /* 346 */
+	{ 5,	0,	printargs,		"SYS_347"	}, /* 347 */
+	{ 5,	0,	printargs,		"SYS_348"	}, /* 348 */
+	{ 5,	0,	printargs,		"SYS_349"	}, /* 349 */
+	{ 5,	0,	printargs,		"SYS_350"	}, /* 350 */
+	{ 5,	0,	printargs,		"SYS_351"	}, /* 351 */
+	{ 5,	0,	printargs,		"SYS_352"	}, /* 352 */
+	{ 5,	0,	printargs,		"SYS_353"	}, /* 353 */
+	{ 5,	0,	printargs,		"SYS_354"	}, /* 354 */
+	{ 5,	0,	printargs,		"SYS_355"	}, /* 355 */
+	{ 5,	0,	printargs,		"SYS_356"	}, /* 356 */
+	{ 5,	0,	printargs,		"SYS_357"	}, /* 357 */
+	{ 5,	0,	printargs,		"SYS_358"	}, /* 358 */
+	{ 5,	0,	printargs,		"SYS_359"	}, /* 359 */
+	{ 5,	0,	printargs,		"SYS_360"	}, /* 360 */
+	{ 5,	0,	printargs,		"SYS_361"	}, /* 361 */
+	{ 5,	0,	printargs,		"SYS_362"	}, /* 362 */
+	{ 5,	0,	printargs,		"SYS_363"	}, /* 363 */
+	{ 5,	0,	printargs,		"SYS_364"	}, /* 364 */
+	{ 5,	0,	printargs,		"SYS_365"	}, /* 365 */
+	{ 5,	0,	printargs,		"SYS_366"	}, /* 366 */
+	{ 5,	0,	printargs,		"SYS_367"	}, /* 367 */
+	{ 5,	0,	printargs,		"SYS_368"	}, /* 368 */
+	{ 5,	0,	printargs,		"SYS_369"	}, /* 369 */
+	{ 5,	0,	printargs,		"SYS_370"	}, /* 370 */
+	{ 5,	0,	printargs,		"SYS_371"	}, /* 371 */
+	{ 5,	0,	printargs,		"SYS_372"	}, /* 372 */
+	{ 5,	0,	printargs,		"SYS_373"	}, /* 373 */
+	{ 5,	0,	printargs,		"SYS_374"	}, /* 374 */
+	{ 5,	0,	printargs,		"SYS_375"	}, /* 375 */
+	{ 5,	0,	printargs,		"SYS_376"	}, /* 376 */
+	{ 5,	0,	printargs,		"SYS_377"	}, /* 377 */
+	{ 5,	0,	printargs,		"SYS_378"	}, /* 378 */
+	{ 5,	0,	printargs,		"SYS_379"	}, /* 379 */
+	{ 5,	0,	printargs,		"SYS_380"	}, /* 380 */
+	{ 5,	0,	printargs,		"SYS_381"	}, /* 381 */
+	{ 5,	0,	printargs,		"SYS_382"	}, /* 382 */
+	{ 5,	0,	printargs,		"SYS_383"	}, /* 383 */
+	{ 5,	0,	printargs,		"SYS_384"	}, /* 384 */
+	{ 5,	0,	printargs,		"SYS_385"	}, /* 385 */
+	{ 5,	0,	printargs,		"SYS_386"	}, /* 386 */
+	{ 5,	0,	printargs,		"SYS_387"	}, /* 387 */
+	{ 5,	0,	printargs,		"SYS_388"	}, /* 388 */
+	{ 5,	0,	printargs,		"SYS_389"	}, /* 389 */
+	{ 5,	0,	printargs,		"SYS_390"	}, /* 390 */
+	{ 5,	0,	printargs,		"SYS_391"	}, /* 391 */
+	{ 5,	0,	printargs,		"SYS_392"	}, /* 392 */
+	{ 5,	0,	printargs,		"SYS_393"	}, /* 393 */
+	{ 5,	0,	printargs,		"SYS_394"	}, /* 394 */
+	{ 5,	0,	printargs,		"SYS_395"	}, /* 395 */
+	{ 5,	0,	printargs,		"SYS_396"	}, /* 396 */
+	{ 5,	0,	printargs,		"SYS_397"	}, /* 397 */
+	{ 5,	0,	printargs,		"SYS_398"	}, /* 398 */
+	{ 5,	0,	printargs,		"SYS_399"	}, /* 399 */
 
-#if SYS_socket_subcall != 300
+#if SYS_socket_subcall != 400
  #error fix me
 #endif
-	{ 8,	0,	printargs,		"socket_subcall"}, /* 300 */
-	{ 3,	TN,	sys_socket,		"socket"	}, /* 301 */
-	{ 3,	TN,	sys_bind,		"bind"		}, /* 302 */
-	{ 3,	TN,	sys_connect,		"connect"	}, /* 303 */
-	{ 2,	TN,	sys_listen,		"listen"	}, /* 304 */
-	{ 3,	TN,	sys_accept,		"accept"	}, /* 305 */
-	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 306 */
-	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 307 */
-	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 308 */
-	{ 4,	TN,	sys_send,		"send", SYS_sub_send }, /* 309 */
-	{ 4,	TN,	sys_recv,		"recv", SYS_sub_recv }, /* 310 */
-	{ 6,	TN,	sys_sendto,		"sendto", SYS_sub_sendto }, /* 311 */
-	{ 6,	TN,	sys_recvfrom,		"recvfrom", SYS_sub_recvfrom }, /* 312 */
-	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 313 */
-	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 314 */
-	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 315 */
-	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 316 */
-	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 317 */
+	{ 8,	0,	printargs,		"socket_subcall"}, /* 400 */
+	{ 3,	TN,	sys_socket,		"socket"	}, /* 401 */
+	{ 3,	TN,	sys_bind,		"bind"		}, /* 402 */
+	{ 3,	TN,	sys_connect,		"connect"	}, /* 403 */
+	{ 2,	TN,	sys_listen,		"listen"	}, /* 404 */
+	{ 3,	TN,	sys_accept,		"accept"	}, /* 405 */
+	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 406 */
+	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 407 */
+	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 408 */
+	{ 4,	TN,	sys_send,		"send", SYS_sub_send }, /* 409 */
+	{ 4,	TN,	sys_recv,		"recv", SYS_sub_recv }, /* 410 */
+	{ 6,	TN,	sys_sendto,		"sendto", SYS_sub_sendto }, /* 411 */
+	{ 6,	TN,	sys_recvfrom,		"recvfrom", SYS_sub_recvfrom }, /* 412 */
+	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 413 */
+	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 414 */
+	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 415 */
+	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 416 */
+	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 417 */
 
-#if SYS_ipc_subcall != 318
+#if SYS_ipc_subcall != 418
  #error fix me
 #endif
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 318 */
-	{ 4,	TI,	sys_semop,		"semop"		}, /* 319 */
-	{ 4,	TI,	sys_semget,		"semget"	}, /* 320 */
-	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 321 */
-	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 322 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 323 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 324 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 325 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 326 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 327 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 328 */
-	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 329 */
-	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 330 */
-	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 331 */
-	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 332 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 333 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 334 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 335 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 336 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 337 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 338 */
-	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 339 */
-	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 340 */
-	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 341 */
-	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 342 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 418 */
+	{ 4,	TI,	sys_semop,		"semop"		}, /* 419 */
+	{ 4,	TI,	sys_semget,		"semget"	}, /* 420 */
+	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 421 */
+	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 422 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 423 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 424 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 425 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 426 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 427 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 428 */
+	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 429 */
+	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 430 */
+	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 431 */
+	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 432 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 433 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 434 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 435 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 436 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 437 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 438 */
+	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 439 */
+	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 440 */
+	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 441 */
+	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 442 */
Index: strace/linux/m68k/syscallent.h
diff -u strace/linux/m68k/syscallent.h:1.2 strace/linux/m68k/syscallent.h:1.3
--- strace/linux/m68k/syscallent.h:1.2	Thu Jan 12 14:34:50 2006
+++ strace/linux/m68k/syscallent.h	Mon Aug 21 23:06:13 2006
@@ -25,7 +25,7 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *
- *	$Id: syscallent.h,v 1.2 2006/01/12 22:34:50 roland Exp $
+ *	$Id: syscallent.h,v 1.3 2006/08/22 06:06:13 roland Exp $
  */
 
 	{ 0,	0,	sys_restart_syscall,	"restart_syscall" }, /* 0 */
@@ -329,54 +329,154 @@
 	{ 5,	0,	printargs,		"SYS_297"	}, /* 297 */
 	{ 5,	0,	printargs,		"SYS_298"	}, /* 298 */
 	{ 5,	0,	printargs,		"SYS_299"	}, /* 299 */
+	{ 5,	0,	printargs,		"SYS_300"	}, /* 300 */
+	{ 5,	0,	printargs,		"SYS_301"	}, /* 301 */
+	{ 5,	0,	printargs,		"SYS_302"	}, /* 302 */
+	{ 5,	0,	printargs,		"SYS_303"	}, /* 303 */
+	{ 5,	0,	printargs,		"SYS_304"	}, /* 304 */
+	{ 5,	0,	printargs,		"SYS_305"	}, /* 305 */
+	{ 5,	0,	printargs,		"SYS_306"	}, /* 306 */
+	{ 5,	0,	printargs,		"SYS_307"	}, /* 307 */
+	{ 5,	0,	printargs,		"SYS_308"	}, /* 308 */
+	{ 5,	0,	printargs,		"SYS_309"	}, /* 309 */
+	{ 5,	0,	printargs,		"SYS_310"	}, /* 310 */
+	{ 5,	0,	printargs,		"SYS_311"	}, /* 311 */
+	{ 5,	0,	printargs,		"SYS_312"	}, /* 312 */
+	{ 5,	0,	printargs,		"SYS_313"	}, /* 313 */
+	{ 5,	0,	printargs,		"SYS_314"	}, /* 314 */
+	{ 5,	0,	printargs,		"SYS_315"	}, /* 315 */
+	{ 5,	0,	printargs,		"SYS_316"	}, /* 316 */
+	{ 5,	0,	printargs,		"SYS_317"	}, /* 317 */
+	{ 5,	0,	printargs,		"SYS_318"	}, /* 318 */
+	{ 5,	0,	printargs,		"SYS_319"	}, /* 319 */
+	{ 5,	0,	printargs,		"SYS_320"	}, /* 320 */
+	{ 5,	0,	printargs,		"SYS_321"	}, /* 321 */
+	{ 5,	0,	printargs,		"SYS_322"	}, /* 322 */
+	{ 5,	0,	printargs,		"SYS_323"	}, /* 323 */
+	{ 5,	0,	printargs,		"SYS_324"	}, /* 324 */
+	{ 5,	0,	printargs,		"SYS_325"	}, /* 325 */
+	{ 5,	0,	printargs,		"SYS_326"	}, /* 326 */
+	{ 5,	0,	printargs,		"SYS_327"	}, /* 327 */
+	{ 5,	0,	printargs,		"SYS_328"	}, /* 328 */
+	{ 5,	0,	printargs,		"SYS_329"	}, /* 329 */
+	{ 5,	0,	printargs,		"SYS_330"	}, /* 330 */
+	{ 5,	0,	printargs,		"SYS_331"	}, /* 331 */
+	{ 5,	0,	printargs,		"SYS_332"	}, /* 332 */
+	{ 5,	0,	printargs,		"SYS_333"	}, /* 333 */
+	{ 5,	0,	printargs,		"SYS_334"	}, /* 334 */
+	{ 5,	0,	printargs,		"SYS_335"	}, /* 335 */
+	{ 5,	0,	printargs,		"SYS_336"	}, /* 336 */
+	{ 5,	0,	printargs,		"SYS_337"	}, /* 337 */
+	{ 5,	0,	printargs,		"SYS_338"	}, /* 338 */
+	{ 5,	0,	printargs,		"SYS_339"	}, /* 339 */
+	{ 5,	0,	printargs,		"SYS_340"	}, /* 340 */
+	{ 5,	0,	printargs,		"SYS_341"	}, /* 341 */
+	{ 5,	0,	printargs,		"SYS_342"	}, /* 342 */
+	{ 5,	0,	printargs,		"SYS_343"	}, /* 343 */
+	{ 5,	0,	printargs,		"SYS_344"	}, /* 344 */
+	{ 5,	0,	printargs,		"SYS_345"	}, /* 345 */
+	{ 5,	0,	printargs,		"SYS_346"	}, /* 346 */
+	{ 5,	0,	printargs,		"SYS_347"	}, /* 347 */
+	{ 5,	0,	printargs,		"SYS_348"	}, /* 348 */
+	{ 5,	0,	printargs,		"SYS_349"	}, /* 349 */
+	{ 5,	0,	printargs,		"SYS_350"	}, /* 350 */
+	{ 5,	0,	printargs,		"SYS_351"	}, /* 351 */
+	{ 5,	0,	printargs,		"SYS_352"	}, /* 352 */
+	{ 5,	0,	printargs,		"SYS_353"	}, /* 353 */
+	{ 5,	0,	printargs,		"SYS_354"	}, /* 354 */
+	{ 5,	0,	printargs,		"SYS_355"	}, /* 355 */
+	{ 5,	0,	printargs,		"SYS_356"	}, /* 356 */
+	{ 5,	0,	printargs,		"SYS_357"	}, /* 357 */
+	{ 5,	0,	printargs,		"SYS_358"	}, /* 358 */
+	{ 5,	0,	printargs,		"SYS_359"	}, /* 359 */
+	{ 5,	0,	printargs,		"SYS_360"	}, /* 360 */
+	{ 5,	0,	printargs,		"SYS_361"	}, /* 361 */
+	{ 5,	0,	printargs,		"SYS_362"	}, /* 362 */
+	{ 5,	0,	printargs,		"SYS_363"	}, /* 363 */
+	{ 5,	0,	printargs,		"SYS_364"	}, /* 364 */
+	{ 5,	0,	printargs,		"SYS_365"	}, /* 365 */
+	{ 5,	0,	printargs,		"SYS_366"	}, /* 366 */
+	{ 5,	0,	printargs,		"SYS_367"	}, /* 367 */
+	{ 5,	0,	printargs,		"SYS_368"	}, /* 368 */
+	{ 5,	0,	printargs,		"SYS_369"	}, /* 369 */
+	{ 5,	0,	printargs,		"SYS_370"	}, /* 370 */
+	{ 5,	0,	printargs,		"SYS_371"	}, /* 371 */
+	{ 5,	0,	printargs,		"SYS_372"	}, /* 372 */
+	{ 5,	0,	printargs,		"SYS_373"	}, /* 373 */
+	{ 5,	0,	printargs,		"SYS_374"	}, /* 374 */
+	{ 5,	0,	printargs,		"SYS_375"	}, /* 375 */
+	{ 5,	0,	printargs,		"SYS_376"	}, /* 376 */
+	{ 5,	0,	printargs,		"SYS_377"	}, /* 377 */
+	{ 5,	0,	printargs,		"SYS_378"	}, /* 378 */
+	{ 5,	0,	printargs,		"SYS_379"	}, /* 379 */
+	{ 5,	0,	printargs,		"SYS_380"	}, /* 380 */
+	{ 5,	0,	printargs,		"SYS_381"	}, /* 381 */
+	{ 5,	0,	printargs,		"SYS_382"	}, /* 382 */
+	{ 5,	0,	printargs,		"SYS_383"	}, /* 383 */
+	{ 5,	0,	printargs,		"SYS_384"	}, /* 384 */
+	{ 5,	0,	printargs,		"SYS_385"	}, /* 385 */
+	{ 5,	0,	printargs,		"SYS_386"	}, /* 386 */
+	{ 5,	0,	printargs,		"SYS_387"	}, /* 387 */
+	{ 5,	0,	printargs,		"SYS_388"	}, /* 388 */
+	{ 5,	0,	printargs,		"SYS_389"	}, /* 389 */
+	{ 5,	0,	printargs,		"SYS_390"	}, /* 390 */
+	{ 5,	0,	printargs,		"SYS_391"	}, /* 391 */
+	{ 5,	0,	printargs,		"SYS_392"	}, /* 392 */
+	{ 5,	0,	printargs,		"SYS_393"	}, /* 393 */
+	{ 5,	0,	printargs,		"SYS_394"	}, /* 394 */
+	{ 5,	0,	printargs,		"SYS_395"	}, /* 395 */
+	{ 5,	0,	printargs,		"SYS_396"	}, /* 396 */
+	{ 5,	0,	printargs,		"SYS_397"	}, /* 397 */
+	{ 5,	0,	printargs,		"SYS_398"	}, /* 398 */
+	{ 5,	0,	printargs,		"SYS_399"	}, /* 399 */
 
-#if SYS_socket_subcall != 300
+#if SYS_socket_subcall != 400
  #error fix me
 #endif
-	{ 8,	0,	printargs,		"socket_subcall"}, /* 300 */
-	{ 3,	TN,	sys_socket,		"socket"	}, /* 301 */
-	{ 3,	TN,	sys_bind,		"bind"		}, /* 302 */
-	{ 3,	TN,	sys_connect,		"connect"	}, /* 303 */
-	{ 2,	TN,	sys_listen,		"listen"	}, /* 304 */
-	{ 3,	TN,	sys_accept,		"accept"	}, /* 305 */
-	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 306 */
-	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 307 */
-	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 308 */
-	{ 4,	TN,	sys_send,		"send", SYS_sub_send }, /* 309 */
-	{ 4,	TN,	sys_recv,		"recv", SYS_sub_recv }, /* 310 */
-	{ 6,	TN,	sys_sendto,		"sendto", SYS_sub_sendto }, /* 311 */
-	{ 6,	TN,	sys_recvfrom,		"recvfrom", SYS_sub_recvfrom }, /* 312 */
-	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 313 */
-	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 314 */
-	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 315 */
-	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 316 */
-	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 317 */
+	{ 8,	0,	printargs,		"socket_subcall"}, /* 400 */
+	{ 3,	TN,	sys_socket,		"socket"	}, /* 401 */
+	{ 3,	TN,	sys_bind,		"bind"		}, /* 402 */
+	{ 3,	TN,	sys_connect,		"connect"	}, /* 403 */
+	{ 2,	TN,	sys_listen,		"listen"	}, /* 404 */
+	{ 3,	TN,	sys_accept,		"accept"	}, /* 405 */
+	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 406 */
+	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 407 */
+	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 408 */
+	{ 4,	TN,	sys_send,		"send", SYS_sub_send }, /* 409 */
+	{ 4,	TN,	sys_recv,		"recv", SYS_sub_recv }, /* 410 */
+	{ 6,	TN,	sys_sendto,		"sendto", SYS_sub_sendto }, /* 411 */
+	{ 6,	TN,	sys_recvfrom,		"recvfrom", SYS_sub_recvfrom }, /* 412 */
+	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 413 */
+	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 414 */
+	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 415 */
+	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 416 */
+	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 417 */
 
-#if SYS_ipc_subcall != 318
+#if SYS_ipc_subcall != 418
  #error fix me
 #endif
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 318 */
-	{ 4,	TI,	sys_semop,		"semop"		}, /* 319 */
-	{ 4,	TI,	sys_semget,		"semget"	}, /* 320 */
-	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 321 */
-	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 322 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 323 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 324 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 325 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 326 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 327 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 328 */
-	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 329 */
-	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 330 */
-	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 331 */
-	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 332 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 333 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 334 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 335 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 336 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 337 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 338 */
-	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 339 */
-	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 340 */
-	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 341 */
-	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 342 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 418 */
+	{ 4,	TI,	sys_semop,		"semop"		}, /* 419 */
+	{ 4,	TI,	sys_semget,		"semget"	}, /* 420 */
+	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 421 */
+	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 422 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 423 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 424 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 425 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 426 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 427 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 428 */
+	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 429 */
+	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 430 */
+	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 431 */
+	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 432 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 433 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 434 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 435 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 436 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 437 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 438 */
+	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 439 */
+	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 440 */
+	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 441 */
+	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 442 */
Index: strace/linux/powerpc/syscallent.h
diff -u strace/linux/powerpc/syscallent.h:1.23 strace/linux/powerpc/syscallent.h:1.24
--- strace/linux/powerpc/syscallent.h:1.23	Thu Jan 12 14:34:50 2006
+++ strace/linux/powerpc/syscallent.h	Mon Aug 21 23:17:33 2006
@@ -25,7 +25,7 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *
- *	$Id: syscallent.h,v 1.23 2006/01/12 22:34:50 roland Exp $
+ *	$Id: syscallent.h,v 1.24 2006/08/22 06:17:33 roland Exp $
  */
 
 	{ 0,	0,	sys_restart_syscall,	"restart_syscall"	}, /* 0 */
@@ -328,54 +328,159 @@
 	{ 5,	0,	printargs,		"SYS_296"		}, /* 296 */
 	{ 5,	0,	printargs,		"SYS_297"		}, /* 297 */
 	{ 5,	0,	printargs,		"SYS_298"		}, /* 298 */
-	{ 5,	0,	printargs,		"SYS_299"		}, /* 299 */
+	{ 5,	0,	printargs,		"SYS_299"		}, /* 299 */	{ 5,	0,	printargs,		"SYS_300"		}, /* 300 */
+	{ 5,	0,	printargs,		"SYS_301"		}, /* 301 */
+	{ 5,	0,	printargs,		"SYS_302"		}, /* 302 */
+	{ 5,	0,	printargs,		"SYS_303"		}, /* 303 */
+	{ 5,	0,	printargs,		"SYS_304"		}, /* 304 */
+	{ 5,	0,	printargs,		"SYS_305"		}, /* 305 */
+	{ 5,	0,	printargs,		"SYS_306"		}, /* 306 */
+	{ 5,	0,	printargs,		"SYS_307"		}, /* 307 */
+	{ 5,	0,	printargs,		"SYS_308"		}, /* 308 */
+	{ 5,	0,	printargs,		"SYS_309"		}, /* 309 */
+	{ 5,	0,	printargs,		"SYS_310"		}, /* 310 */
+	{ 5,	0,	printargs,		"SYS_311"		}, /* 311 */
+	{ 5,	0,	printargs,		"SYS_312"		}, /* 312 */
+	{ 5,	0,	printargs,		"SYS_313"		}, /* 313 */
+	{ 5,	0,	printargs,		"SYS_314"		}, /* 314 */
+	{ 5,	0,	printargs,		"SYS_315"		}, /* 315 */
+	{ 5,	0,	printargs,		"SYS_316"		}, /* 316 */
+	{ 5,	0,	printargs,		"SYS_317"		}, /* 317 */
+	{ 5,	0,	printargs,		"SYS_318"		}, /* 318 */
+	{ 5,	0,	printargs,		"SYS_319"		}, /* 319 */
+	{ 5,	0,	printargs,		"SYS_320"		}, /* 320 */
+	{ 5,	0,	printargs,		"SYS_321"		}, /* 321 */
+	{ 5,	0,	printargs,		"SYS_322"		}, /* 322 */
+	{ 5,	0,	printargs,		"SYS_323"		}, /* 323 */
+	{ 5,	0,	printargs,		"SYS_324"		}, /* 324 */
+	{ 5,	0,	printargs,		"SYS_325"		}, /* 325 */
+	{ 5,	0,	printargs,		"SYS_326"		}, /* 326 */
+	{ 5,	0,	printargs,		"SYS_327"		}, /* 327 */
+	{ 5,	0,	printargs,		"SYS_328"		}, /* 328 */
+	{ 5,	0,	printargs,		"SYS_329"		}, /* 329 */
+	{ 5,	0,	printargs,		"SYS_330"		}, /* 330 */
+	{ 5,	0,	printargs,		"SYS_331"		}, /* 331 */
+	{ 5,	0,	printargs,		"SYS_332"		}, /* 332 */
+	{ 5,	0,	printargs,		"SYS_333"		}, /* 333 */
+	{ 5,	0,	printargs,		"SYS_334"		}, /* 334 */
+	{ 5,	0,	printargs,		"SYS_335"		}, /* 335 */
+	{ 5,	0,	printargs,		"SYS_336"		}, /* 336 */
+	{ 5,	0,	printargs,		"SYS_337"		}, /* 337 */
+	{ 5,	0,	printargs,		"SYS_338"		}, /* 338 */
+	{ 5,	0,	printargs,		"SYS_339"		}, /* 339 */
+	{ 5,	0,	printargs,		"SYS_340"		}, /* 340 */
+	{ 5,	0,	printargs,		"SYS_341"		}, /* 341 */
+	{ 5,	0,	printargs,		"SYS_342"		}, /* 342 */
+	{ 5,	0,	printargs,		"SYS_343"		}, /* 343 */
+	{ 5,	0,	printargs,		"SYS_344"		}, /* 344 */
+	{ 5,	0,	printargs,		"SYS_345"		}, /* 345 */
+	{ 5,	0,	printargs,		"SYS_346"		}, /* 346 */
+	{ 5,	0,	printargs,		"SYS_347"		}, /* 347 */
+	{ 5,	0,	printargs,		"SYS_348"		}, /* 348 */
+	{ 5,	0,	printargs,		"SYS_349"		}, /* 349 */
+	{ 5,	0,	printargs,		"SYS_350"		}, /* 350 */
+	{ 5,	0,	printargs,		"SYS_351"		}, /* 351 */
+	{ 5,	0,	printargs,		"SYS_352"		}, /* 352 */
+	{ 5,	0,	printargs,		"SYS_353"		}, /* 353 */
+	{ 5,	0,	printargs,		"SYS_354"		}, /* 354 */
+	{ 5,	0,	printargs,		"SYS_355"		}, /* 355 */
+	{ 5,	0,	printargs,		"SYS_356"		}, /* 356 */
+	{ 5,	0,	printargs,		"SYS_357"		}, /* 357 */
+	{ 5,	0,	printargs,		"SYS_358"		}, /* 358 */
+	{ 5,	0,	printargs,		"SYS_359"		}, /* 359 */
+	{ 5,	0,	printargs,		"SYS_360"		}, /* 360 */
+	{ 5,	0,	printargs,		"SYS_361"		}, /* 361 */
+	{ 5,	0,	printargs,		"SYS_362"		}, /* 362 */
+	{ 5,	0,	printargs,		"SYS_363"		}, /* 363 */
+	{ 5,	0,	printargs,		"SYS_364"		}, /* 364 */
+	{ 5,	0,	printargs,		"SYS_365"		}, /* 365 */
+	{ 5,	0,	printargs,		"SYS_366"		}, /* 366 */
+	{ 5,	0,	printargs,		"SYS_367"		}, /* 367 */
+	{ 5,	0,	printargs,		"SYS_368"		}, /* 368 */
+	{ 5,	0,	printargs,		"SYS_369"		}, /* 369 */
+	{ 5,	0,	printargs,		"SYS_370"		}, /* 370 */
+	{ 5,	0,	printargs,		"SYS_371"		}, /* 371 */
+	{ 5,	0,	printargs,		"SYS_372"		}, /* 372 */
+	{ 5,	0,	printargs,		"SYS_373"		}, /* 373 */
+	{ 5,	0,	printargs,		"SYS_374"		}, /* 374 */
+	{ 5,	0,	printargs,		"SYS_375"		}, /* 375 */
+	{ 5,	0,	printargs,		"SYS_376"		}, /* 376 */
+	{ 5,	0,	printargs,		"SYS_377"		}, /* 377 */
+	{ 5,	0,	printargs,		"SYS_378"		}, /* 378 */
+	{ 5,	0,	printargs,		"SYS_379"		}, /* 379 */
+	{ 5,	0,	printargs,		"SYS_380"		}, /* 380 */
+	{ 5,	0,	printargs,		"SYS_381"		}, /* 381 */
+	{ 5,	0,	printargs,		"SYS_382"		}, /* 382 */
+	{ 5,	0,	printargs,		"SYS_383"		}, /* 383 */
+	{ 5,	0,	printargs,		"SYS_384"		}, /* 384 */
+	{ 5,	0,	printargs,		"SYS_385"		}, /* 385 */
+	{ 5,	0,	printargs,		"SYS_386"		}, /* 386 */
+	{ 5,	0,	printargs,		"SYS_387"		}, /* 387 */
+	{ 5,	0,	printargs,		"SYS_388"		}, /* 388 */
+	{ 5,	0,	printargs,		"SYS_389"		}, /* 389 */
+	{ 5,	0,	printargs,		"SYS_390"		}, /* 390 */
+	{ 5,	0,	printargs,		"SYS_391"		}, /* 391 */
+	{ 5,	0,	printargs,		"SYS_392"		}, /* 392 */
+	{ 5,	0,	printargs,		"SYS_393"		}, /* 393 */
+	{ 5,	0,	printargs,		"SYS_394"		}, /* 394 */
+	{ 5,	0,	printargs,		"SYS_395"		}, /* 395 */
+	{ 5,	0,	printargs,		"SYS_396"		}, /* 396 */
+	{ 5,	0,	printargs,		"SYS_397"		}, /* 397 */
+	{ 5,	0,	printargs,		"SYS_398"		}, /* 398 */
+	{ 5,	0,	printargs,		"SYS_399"		}, /* 399 */
 
-	{ 8,	0,	printargs,		"socket_subcall"}, /* 300 */
-	{ 3,	TN,	sys_socket,		"socket"	}, /* 301 */
-	{ 3,	TN,	sys_bind,		"bind"		}, /* 302 */
-	{ 3,	TN,	sys_connect,		"connect"	}, /* 303 */
-	{ 2,	TN,	sys_listen,		"listen"	}, /* 304 */
-	{ 3,	TN,	sys_accept,		"accept"	}, /* 305 */
-	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 306 */
-	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 307 */
-	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 308 */
-	{ 4,	TN,	sys_send,		"send"		}, /* 309 */
-	{ 4,	TN,	sys_recv,		"recv"		}, /* 310 */
-	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 311 */
-	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 312 */
-	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 313 */
-	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 314 */
-	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 315 */
-	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 316 */
-	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 317 */
+#if SYS_socket_subcall != 400
+ #error fix me
+#endif
+	{ 8,	0,	printargs,		"socket_subcall"}, /* 400 */
+	{ 3,	TN,	sys_socket,		"socket"	}, /* 401 */
+	{ 3,	TN,	sys_bind,		"bind"		}, /* 402 */
+	{ 3,	TN,	sys_connect,		"connect"	}, /* 403 */
+	{ 2,	TN,	sys_listen,		"listen"	}, /* 404 */
+	{ 3,	TN,	sys_accept,		"accept"	}, /* 405 */
+	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 406 */
+	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 407 */
+	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 408 */
+	{ 4,	TN,	sys_send,		"send"		}, /* 409 */
+	{ 4,	TN,	sys_recv,		"recv"		}, /* 410 */
+	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 411 */
+	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 412 */
+	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 413 */
+	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 414 */
+	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 415 */
+	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 416 */
+	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 417 */
 
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 318 */
-	{ 4,	TI,	printargs,		"semop"		}, /* 319 */
-	{ 4,	TI,	sys_semget,		"semget"	}, /* 320 */
-	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 321 */
-	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 322 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 323 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 324 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 325 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 326 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 327 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 328 */
-	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 329 */
-	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 330 */
-	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 331 */
-	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 332 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 333 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 334 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 335 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 336 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 337 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 338 */
-	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 339 */
-	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 340 */
-	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 341 */
-	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 342 */
-	{ 5,	0,	printargs,		"SYS_343"	}, /* 343 */
-	{ 5,	0,	printargs,		"SYS_344"	}, /* 344 */
-	{ 5,	0,	printargs,		"SYS_345"	}, /* 345 */
-	{ 5,	0,	printargs,		"SYS_346"	}, /* 346 */
-	{ 5,	0,	printargs,		"SYS_347"	}, /* 347 */
+#if SYS_ipc_subcall != 418
+ #error fix me
+#endif
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 418 */
+	{ 4,	TI,	printargs,		"semop"		}, /* 419 */
+	{ 4,	TI,	sys_semget,		"semget"	}, /* 420 */
+	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 421 */
+	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 422 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 423 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 424 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 425 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 426 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 427 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 428 */
+	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 429 */
+	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 430 */
+	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 431 */
+	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 432 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 433 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 434 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 435 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 436 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 437 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 438 */
+	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 439 */
+	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 440 */
+	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 441 */
+	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 442 */
+	{ 5,	0,	printargs,		"SYS_343"	}, /* 443 */
+	{ 5,	0,	printargs,		"SYS_344"	}, /* 444 */
+	{ 5,	0,	printargs,		"SYS_345"	}, /* 445 */
+	{ 5,	0,	printargs,		"SYS_346"	}, /* 446 */
+	{ 5,	0,	printargs,		"SYS_347"	}, /* 447 */
Index: strace/linux/s390/syscallent.h
diff -u strace/linux/s390/syscallent.h:1.13 strace/linux/s390/syscallent.h:1.14
--- strace/linux/s390/syscallent.h:1.13	Thu Dec  1 20:19:13 2005
+++ strace/linux/s390/syscallent.h	Mon Aug 21 23:06:13 2006
@@ -329,54 +329,154 @@
 	{ 5,	0,	printargs,		"SYS_297"	}, /* 297 */
 	{ 5,	0,	printargs,		"SYS_298"	}, /* 298 */
 	{ 5,	0,	printargs,		"SYS_299"	}, /* 299 */
+	{ 5,	0,	printargs,		"SYS_300"	}, /* 300 */
+	{ 5,	0,	printargs,		"SYS_301"	}, /* 301 */
+	{ 5,	0,	printargs,		"SYS_302"	}, /* 302 */
+	{ 5,	0,	printargs,		"SYS_303"	}, /* 303 */
+	{ 5,	0,	printargs,		"SYS_304"	}, /* 304 */
+	{ 5,	0,	printargs,		"SYS_305"	}, /* 305 */
+	{ 5,	0,	printargs,		"SYS_306"	}, /* 306 */
+	{ 5,	0,	printargs,		"SYS_307"	}, /* 307 */
+	{ 5,	0,	printargs,		"SYS_308"	}, /* 308 */
+	{ 5,	0,	printargs,		"SYS_309"	}, /* 309 */
+	{ 5,	0,	printargs,		"SYS_310"	}, /* 310 */
+	{ 5,	0,	printargs,		"SYS_311"	}, /* 311 */
+	{ 5,	0,	printargs,		"SYS_312"	}, /* 312 */
+	{ 5,	0,	printargs,		"SYS_313"	}, /* 313 */
+	{ 5,	0,	printargs,		"SYS_314"	}, /* 314 */
+	{ 5,	0,	printargs,		"SYS_315"	}, /* 315 */
+	{ 5,	0,	printargs,		"SYS_316"	}, /* 316 */
+	{ 5,	0,	printargs,		"SYS_317"	}, /* 317 */
+	{ 5,	0,	printargs,		"SYS_318"	}, /* 318 */
+	{ 5,	0,	printargs,		"SYS_319"	}, /* 319 */
+	{ 5,	0,	printargs,		"SYS_320"	}, /* 320 */
+	{ 5,	0,	printargs,		"SYS_321"	}, /* 321 */
+	{ 5,	0,	printargs,		"SYS_322"	}, /* 322 */
+	{ 5,	0,	printargs,		"SYS_323"	}, /* 323 */
+	{ 5,	0,	printargs,		"SYS_324"	}, /* 324 */
+	{ 5,	0,	printargs,		"SYS_325"	}, /* 325 */
+	{ 5,	0,	printargs,		"SYS_326"	}, /* 326 */
+	{ 5,	0,	printargs,		"SYS_327"	}, /* 327 */
+	{ 5,	0,	printargs,		"SYS_328"	}, /* 328 */
+	{ 5,	0,	printargs,		"SYS_329"	}, /* 329 */
+	{ 5,	0,	printargs,		"SYS_330"	}, /* 330 */
+	{ 5,	0,	printargs,		"SYS_331"	}, /* 331 */
+	{ 5,	0,	printargs,		"SYS_332"	}, /* 332 */
+	{ 5,	0,	printargs,		"SYS_333"	}, /* 333 */
+	{ 5,	0,	printargs,		"SYS_334"	}, /* 334 */
+	{ 5,	0,	printargs,		"SYS_335"	}, /* 335 */
+	{ 5,	0,	printargs,		"SYS_336"	}, /* 336 */
+	{ 5,	0,	printargs,		"SYS_337"	}, /* 337 */
+	{ 5,	0,	printargs,		"SYS_338"	}, /* 338 */
+	{ 5,	0,	printargs,		"SYS_339"	}, /* 339 */
+	{ 5,	0,	printargs,		"SYS_340"	}, /* 340 */
+	{ 5,	0,	printargs,		"SYS_341"	}, /* 341 */
+	{ 5,	0,	printargs,		"SYS_342"	}, /* 342 */
+	{ 5,	0,	printargs,		"SYS_343"	}, /* 343 */
+	{ 5,	0,	printargs,		"SYS_344"	}, /* 344 */
+	{ 5,	0,	printargs,		"SYS_345"	}, /* 345 */
+	{ 5,	0,	printargs,		"SYS_346"	}, /* 346 */
+	{ 5,	0,	printargs,		"SYS_347"	}, /* 347 */
+	{ 5,	0,	printargs,		"SYS_348"	}, /* 348 */
+	{ 5,	0,	printargs,		"SYS_349"	}, /* 349 */
+	{ 5,	0,	printargs,		"SYS_350"	}, /* 350 */
+	{ 5,	0,	printargs,		"SYS_351"	}, /* 351 */
+	{ 5,	0,	printargs,		"SYS_352"	}, /* 352 */
+	{ 5,	0,	printargs,		"SYS_353"	}, /* 353 */
+	{ 5,	0,	printargs,		"SYS_354"	}, /* 354 */
+	{ 5,	0,	printargs,		"SYS_355"	}, /* 355 */
+	{ 5,	0,	printargs,		"SYS_356"	}, /* 356 */
+	{ 5,	0,	printargs,		"SYS_357"	}, /* 357 */
+	{ 5,	0,	printargs,		"SYS_358"	}, /* 358 */
+	{ 5,	0,	printargs,		"SYS_359"	}, /* 359 */
+	{ 5,	0,	printargs,		"SYS_360"	}, /* 360 */
+	{ 5,	0,	printargs,		"SYS_361"	}, /* 361 */
+	{ 5,	0,	printargs,		"SYS_362"	}, /* 362 */
+	{ 5,	0,	printargs,		"SYS_363"	}, /* 363 */
+	{ 5,	0,	printargs,		"SYS_364"	}, /* 364 */
+	{ 5,	0,	printargs,		"SYS_365"	}, /* 365 */
+	{ 5,	0,	printargs,		"SYS_366"	}, /* 366 */
+	{ 5,	0,	printargs,		"SYS_367"	}, /* 367 */
+	{ 5,	0,	printargs,		"SYS_368"	}, /* 368 */
+	{ 5,	0,	printargs,		"SYS_369"	}, /* 369 */
+	{ 5,	0,	printargs,		"SYS_370"	}, /* 370 */
+	{ 5,	0,	printargs,		"SYS_371"	}, /* 371 */
+	{ 5,	0,	printargs,		"SYS_372"	}, /* 372 */
+	{ 5,	0,	printargs,		"SYS_373"	}, /* 373 */
+	{ 5,	0,	printargs,		"SYS_374"	}, /* 374 */
+	{ 5,	0,	printargs,		"SYS_375"	}, /* 375 */
+	{ 5,	0,	printargs,		"SYS_376"	}, /* 376 */
+	{ 5,	0,	printargs,		"SYS_377"	}, /* 377 */
+	{ 5,	0,	printargs,		"SYS_378"	}, /* 378 */
+	{ 5,	0,	printargs,		"SYS_379"	}, /* 379 */
+	{ 5,	0,	printargs,		"SYS_380"	}, /* 380 */
+	{ 5,	0,	printargs,		"SYS_381"	}, /* 381 */
+	{ 5,	0,	printargs,		"SYS_382"	}, /* 382 */
+	{ 5,	0,	printargs,		"SYS_383"	}, /* 383 */
+	{ 5,	0,	printargs,		"SYS_384"	}, /* 384 */
+	{ 5,	0,	printargs,		"SYS_385"	}, /* 385 */
+	{ 5,	0,	printargs,		"SYS_386"	}, /* 386 */
+	{ 5,	0,	printargs,		"SYS_387"	}, /* 387 */
+	{ 5,	0,	printargs,		"SYS_388"	}, /* 388 */
+	{ 5,	0,	printargs,		"SYS_389"	}, /* 389 */
+	{ 5,	0,	printargs,		"SYS_390"	}, /* 390 */
+	{ 5,	0,	printargs,		"SYS_391"	}, /* 391 */
+	{ 5,	0,	printargs,		"SYS_392"	}, /* 392 */
+	{ 5,	0,	printargs,		"SYS_393"	}, /* 393 */
+	{ 5,	0,	printargs,		"SYS_394"	}, /* 394 */
+	{ 5,	0,	printargs,		"SYS_395"	}, /* 395 */
+	{ 5,	0,	printargs,		"SYS_396"	}, /* 396 */
+	{ 5,	0,	printargs,		"SYS_397"	}, /* 397 */
+	{ 5,	0,	printargs,		"SYS_398"	}, /* 398 */
+	{ 5,	0,	printargs,		"SYS_399"	}, /* 399 */
 
-#if SYS_socket_subcall != 300
+#if SYS_socket_subcall != 400
  #error fix me
 #endif
-	{ 8,	0,	printargs,		"socket_subcall"}, /* 300 */
-	{ 3,	TN,	sys_socket,		"socket"	}, /* 301 */
-	{ 3,	TN,	sys_bind,		"bind"		}, /* 302 */
-	{ 3,	TN,	sys_connect,		"connect"	}, /* 303 */
-	{ 2,	TN,	sys_listen,		"listen"	}, /* 304 */
-	{ 3,	TN,	sys_accept,		"accept"	}, /* 305 */
-	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 306 */
-	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 307 */
-	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 308 */
-	{ 4,	TN,	sys_send,		"send"		}, /* 309 */
-	{ 4,	TN,	sys_recv,		"recv"		}, /* 310 */
-	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 311 */
-	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 312 */
-	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 313 */
-	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 314 */
-	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 315 */
-	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 316 */
-	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 317 */
+	{ 8,	0,	printargs,		"socket_subcall"}, /* 400 */
+	{ 3,	TN,	sys_socket,		"socket"	}, /* 401 */
+	{ 3,	TN,	sys_bind,		"bind"		}, /* 402 */
+	{ 3,	TN,	sys_connect,		"connect"	}, /* 403 */
+	{ 2,	TN,	sys_listen,		"listen"	}, /* 404 */
+	{ 3,	TN,	sys_accept,		"accept"	}, /* 405 */
+	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 406 */
+	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 407 */
+	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 408 */
+	{ 4,	TN,	sys_send,		"send"		}, /* 409 */
+	{ 4,	TN,	sys_recv,		"recv"		}, /* 410 */
+	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 411 */
+	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 412 */
+	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 413 */
+	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 414 */
+	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 415 */
+	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 416 */
+	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 417 */
 
-#if SYS_ipc_subcall != 318
+#if SYS_ipc_subcall != 418
  #error fix me
 #endif
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 318 */
-	{ 4,	TI,	sys_semop,		"semop"		}, /* 319 */
-	{ 4,	TI,	sys_semget,		"semget"	}, /* 320 */
-	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 321 */
-	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 322 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 323 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 324 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 325 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 326 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 327 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 328 */
-	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 329 */
-	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 330 */
-	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 331 */
-	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 332 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 333 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 334 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 335 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 336 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 337 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 338 */
-	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 339 */
-	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 340 */
-	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 341 */
-	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 342 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 418 */
+	{ 4,	TI,	sys_semop,		"semop"		}, /* 419 */
+	{ 4,	TI,	sys_semget,		"semget"	}, /* 420 */
+	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 421 */
+	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 422 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 423 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 424 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 425 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 426 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 427 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 428 */
+	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 429 */
+	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 430 */
+	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 431 */
+	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 432 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 433 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 434 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 435 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 436 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 437 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 438 */
+	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 439 */
+	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 440 */
+	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 441 */
+	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 442 */
Index: strace/linux/s390x/syscallent.h
diff -u strace/linux/s390x/syscallent.h:1.12 strace/linux/s390x/syscallent.h:1.13
--- strace/linux/s390x/syscallent.h:1.12	Thu Dec  1 20:19:13 2005
+++ strace/linux/s390x/syscallent.h	Mon Aug 21 23:06:13 2006
@@ -328,54 +328,154 @@
 	{ 5,	0,	printargs,		"SYS_297"	}, /* 297 */
 	{ 5,	0,	printargs,		"SYS_298"	}, /* 298 */
 	{ 5,	0,	printargs,		"SYS_299"	}, /* 299 */
+	{ 5,	0,	printargs,		"SYS_300"	}, /* 300 */
+	{ 5,	0,	printargs,		"SYS_301"	}, /* 301 */
+	{ 5,	0,	printargs,		"SYS_302"	}, /* 302 */
+	{ 5,	0,	printargs,		"SYS_303"	}, /* 303 */
+	{ 5,	0,	printargs,		"SYS_304"	}, /* 304 */
+	{ 5,	0,	printargs,		"SYS_305"	}, /* 305 */
+	{ 5,	0,	printargs,		"SYS_306"	}, /* 306 */
+	{ 5,	0,	printargs,		"SYS_307"	}, /* 307 */
+	{ 5,	0,	printargs,		"SYS_308"	}, /* 308 */
+	{ 5,	0,	printargs,		"SYS_309"	}, /* 309 */
+	{ 5,	0,	printargs,		"SYS_310"	}, /* 310 */
+	{ 5,	0,	printargs,		"SYS_311"	}, /* 311 */
+	{ 5,	0,	printargs,		"SYS_312"	}, /* 312 */
+	{ 5,	0,	printargs,		"SYS_313"	}, /* 313 */
+	{ 5,	0,	printargs,		"SYS_314"	}, /* 314 */
+	{ 5,	0,	printargs,		"SYS_315"	}, /* 315 */
+	{ 5,	0,	printargs,		"SYS_316"	}, /* 316 */
+	{ 5,	0,	printargs,		"SYS_317"	}, /* 317 */
+	{ 5,	0,	printargs,		"SYS_318"	}, /* 318 */
+	{ 5,	0,	printargs,		"SYS_319"	}, /* 319 */
+	{ 5,	0,	printargs,		"SYS_320"	}, /* 320 */
+	{ 5,	0,	printargs,		"SYS_321"	}, /* 321 */
+	{ 5,	0,	printargs,		"SYS_322"	}, /* 322 */
+	{ 5,	0,	printargs,		"SYS_323"	}, /* 323 */
+	{ 5,	0,	printargs,		"SYS_324"	}, /* 324 */
+	{ 5,	0,	printargs,		"SYS_325"	}, /* 325 */
+	{ 5,	0,	printargs,		"SYS_326"	}, /* 326 */
+	{ 5,	0,	printargs,		"SYS_327"	}, /* 327 */
+	{ 5,	0,	printargs,		"SYS_328"	}, /* 328 */
+	{ 5,	0,	printargs,		"SYS_329"	}, /* 329 */
+	{ 5,	0,	printargs,		"SYS_330"	}, /* 330 */
+	{ 5,	0,	printargs,		"SYS_331"	}, /* 331 */
+	{ 5,	0,	printargs,		"SYS_332"	}, /* 332 */
+	{ 5,	0,	printargs,		"SYS_333"	}, /* 333 */
+	{ 5,	0,	printargs,		"SYS_334"	}, /* 334 */
+	{ 5,	0,	printargs,		"SYS_335"	}, /* 335 */
+	{ 5,	0,	printargs,		"SYS_336"	}, /* 336 */
+	{ 5,	0,	printargs,		"SYS_337"	}, /* 337 */
+	{ 5,	0,	printargs,		"SYS_338"	}, /* 338 */
+	{ 5,	0,	printargs,		"SYS_339"	}, /* 339 */
+	{ 5,	0,	printargs,		"SYS_340"	}, /* 340 */
+	{ 5,	0,	printargs,		"SYS_341"	}, /* 341 */
+	{ 5,	0,	printargs,		"SYS_342"	}, /* 342 */
+	{ 5,	0,	printargs,		"SYS_343"	}, /* 343 */
+	{ 5,	0,	printargs,		"SYS_344"	}, /* 344 */
+	{ 5,	0,	printargs,		"SYS_345"	}, /* 345 */
+	{ 5,	0,	printargs,		"SYS_346"	}, /* 346 */
+	{ 5,	0,	printargs,		"SYS_347"	}, /* 347 */
+	{ 5,	0,	printargs,		"SYS_348"	}, /* 348 */
+	{ 5,	0,	printargs,		"SYS_349"	}, /* 349 */
+	{ 5,	0,	printargs,		"SYS_350"	}, /* 350 */
+	{ 5,	0,	printargs,		"SYS_351"	}, /* 351 */
+	{ 5,	0,	printargs,		"SYS_352"	}, /* 352 */
+	{ 5,	0,	printargs,		"SYS_353"	}, /* 353 */
+	{ 5,	0,	printargs,		"SYS_354"	}, /* 354 */
+	{ 5,	0,	printargs,		"SYS_355"	}, /* 355 */
+	{ 5,	0,	printargs,		"SYS_356"	}, /* 356 */
+	{ 5,	0,	printargs,		"SYS_357"	}, /* 357 */
+	{ 5,	0,	printargs,		"SYS_358"	}, /* 358 */
+	{ 5,	0,	printargs,		"SYS_359"	}, /* 359 */
+	{ 5,	0,	printargs,		"SYS_360"	}, /* 360 */
+	{ 5,	0,	printargs,		"SYS_361"	}, /* 361 */
+	{ 5,	0,	printargs,		"SYS_362"	}, /* 362 */
+	{ 5,	0,	printargs,		"SYS_363"	}, /* 363 */
+	{ 5,	0,	printargs,		"SYS_364"	}, /* 364 */
+	{ 5,	0,	printargs,		"SYS_365"	}, /* 365 */
+	{ 5,	0,	printargs,		"SYS_366"	}, /* 366 */
+	{ 5,	0,	printargs,		"SYS_367"	}, /* 367 */
+	{ 5,	0,	printargs,		"SYS_368"	}, /* 368 */
+	{ 5,	0,	printargs,		"SYS_369"	}, /* 369 */
+	{ 5,	0,	printargs,		"SYS_370"	}, /* 370 */
+	{ 5,	0,	printargs,		"SYS_371"	}, /* 371 */
+	{ 5,	0,	printargs,		"SYS_372"	}, /* 372 */
+	{ 5,	0,	printargs,		"SYS_373"	}, /* 373 */
+	{ 5,	0,	printargs,		"SYS_374"	}, /* 374 */
+	{ 5,	0,	printargs,		"SYS_375"	}, /* 375 */
+	{ 5,	0,	printargs,		"SYS_376"	}, /* 376 */
+	{ 5,	0,	printargs,		"SYS_377"	}, /* 377 */
+	{ 5,	0,	printargs,		"SYS_378"	}, /* 378 */
+	{ 5,	0,	printargs,		"SYS_379"	}, /* 379 */
+	{ 5,	0,	printargs,		"SYS_380"	}, /* 380 */
+	{ 5,	0,	printargs,		"SYS_381"	}, /* 381 */
+	{ 5,	0,	printargs,		"SYS_382"	}, /* 382 */
+	{ 5,	0,	printargs,		"SYS_383"	}, /* 383 */
+	{ 5,	0,	printargs,		"SYS_384"	}, /* 384 */
+	{ 5,	0,	printargs,		"SYS_385"	}, /* 385 */
+	{ 5,	0,	printargs,		"SYS_386"	}, /* 386 */
+	{ 5,	0,	printargs,		"SYS_387"	}, /* 387 */
+	{ 5,	0,	printargs,		"SYS_388"	}, /* 388 */
+	{ 5,	0,	printargs,		"SYS_389"	}, /* 389 */
+	{ 5,	0,	printargs,		"SYS_390"	}, /* 390 */
+	{ 5,	0,	printargs,		"SYS_391"	}, /* 391 */
+	{ 5,	0,	printargs,		"SYS_392"	}, /* 392 */
+	{ 5,	0,	printargs,		"SYS_393"	}, /* 393 */
+	{ 5,	0,	printargs,		"SYS_394"	}, /* 394 */
+	{ 5,	0,	printargs,		"SYS_395"	}, /* 395 */
+	{ 5,	0,	printargs,		"SYS_396"	}, /* 396 */
+	{ 5,	0,	printargs,		"SYS_397"	}, /* 397 */
+	{ 5,	0,	printargs,		"SYS_398"	}, /* 398 */
+	{ 5,	0,	printargs,		"SYS_399"	}, /* 399 */
 
-#if SYS_socket_subcall != 300
+#if SYS_socket_subcall != 400
  #error fix me
 #endif
-	{ 8,	0,	printargs,		"socket_subcall"}, /* 300 */
-	{ 3,	TN,	sys_socket,		"socket"	}, /* 301 */
-	{ 3,	TN,	sys_bind,		"bind"		}, /* 302 */
-	{ 3,	TN,	sys_connect,		"connect"	}, /* 303 */
-	{ 2,	TN,	sys_listen,		"listen"	}, /* 304 */
-	{ 3,	TN,	sys_accept,		"accept"	}, /* 305 */
-	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 306 */
-	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 307 */
-	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 308 */
-	{ 4,	TN,	sys_send,		"send"		}, /* 309 */
-	{ 4,	TN,	sys_recv,		"recv"		}, /* 310 */
-	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 311 */
-	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 312 */
-	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 313 */
-	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 314 */
-	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 315 */
-	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 316 */
-	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 317 */
+	{ 8,	0,	printargs,		"socket_subcall"}, /* 400 */
+	{ 3,	TN,	sys_socket,		"socket"	}, /* 401 */
+	{ 3,	TN,	sys_bind,		"bind"		}, /* 402 */
+	{ 3,	TN,	sys_connect,		"connect"	}, /* 403 */
+	{ 2,	TN,	sys_listen,		"listen"	}, /* 404 */
+	{ 3,	TN,	sys_accept,		"accept"	}, /* 405 */
+	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 406 */
+	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 407 */
+	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 408 */
+	{ 4,	TN,	sys_send,		"send"		}, /* 409 */
+	{ 4,	TN,	sys_recv,		"recv"		}, /* 410 */
+	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 411 */
+	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 412 */
+	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 413 */
+	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 414 */
+	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 415 */
+	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 416 */
+	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 417 */
 
-#if SYS_ipc_subcall != 318
+#if SYS_ipc_subcall != 418
  #error fix me
 #endif
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 318 */
-	{ 4,	TI,	sys_semop,		"semop"		}, /* 319 */
-	{ 4,	TI,	sys_semget,		"semget"	}, /* 320 */
-	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 321 */
-	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 322 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 323 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 324 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 325 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 326 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 327 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 328 */
-	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 329 */
-	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 330 */
-	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 331 */
-	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 332 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 333 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 334 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 335 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 336 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 337 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 338 */
-	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 339 */
-	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 340 */
-	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 341 */
-	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 342 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 418 */
+	{ 4,	TI,	sys_semop,		"semop"		}, /* 419 */
+	{ 4,	TI,	sys_semget,		"semget"	}, /* 420 */
+	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 421 */
+	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 422 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 423 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 424 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 425 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 426 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 427 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 428 */
+	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 429 */
+	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 430 */
+	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 431 */
+	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 432 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 433 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 434 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 435 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 436 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 437 */
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 438 */
+	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 439 */
+	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 440 */
+	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 441 */
+	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 442 */
Index: strace/linux/sh/syscallent.h
diff -u strace/linux/sh/syscallent.h:1.8 strace/linux/sh/syscallent.h:1.9
--- strace/linux/sh/syscallent.h:1.8	Thu Jan 12 14:34:50 2006
+++ strace/linux/sh/syscallent.h	Mon Aug 21 23:17:33 2006
@@ -27,7 +27,7 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *
- *	$Id: syscallent.h,v 1.8 2006/01/12 22:34:50 roland Exp $
+ *	$Id: syscallent.h,v 1.9 2006/08/22 06:17:33 roland Exp $
  */
 
 	{ 0,	0,	sys_restart_syscall,	"restart_syscall"}, /* 0 */
@@ -314,24 +314,150 @@
 	{ 5,	0,	printargs,		"SYS_277"	}, /* 277 */
 	{ 5,	0,	printargs,		"SYS_278"	}, /* 278 */
 	{ 5,	0,	printargs,		"SYS_279"	}, /* 279 */
-
-	{ 8,	0,	printargs,		"socket_subcall"}, /* 280 */
-	{ 3,	TN,	sys_socket,		"socket"	}, /* 281 */
-	{ 3,	TN,	sys_bind,		"bind"		}, /* 282 */
-	{ 3,	TN,	sys_connect,		"connect"	}, /* 283 */
-	{ 2,	TN,	sys_listen,		"listen"	}, /* 284 */
-	{ 3,	TN,	sys_accept,		"accept"	}, /* 285 */
-	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 286 */
-	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 287 */
-	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 288 */
-	{ 4,	TN,	sys_send,		"send"		}, /* 289 */
-	{ 4,	TN,	sys_recv,		"recv"		}, /* 290 */
-	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 291 */
-	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 292 */
-	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 293 */
-	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 294 */
-	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 295 */
-	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 296 */
-	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 297 */
+	{ 5,	0,	printargs,		"SYS_280"	}, /* 280 */
+	{ 5,	0,	printargs,		"SYS_281"	}, /* 281 */
+	{ 5,	0,	printargs,		"SYS_282"	}, /* 282 */
+	{ 5,	0,	printargs,		"SYS_283"	}, /* 283 */
+	{ 5,	0,	printargs,		"SYS_284"	}, /* 284 */
+	{ 5,	0,	printargs,		"SYS_285"	}, /* 285 */
+	{ 5,	0,	printargs,		"SYS_286"	}, /* 286 */
+	{ 5,	0,	printargs,		"SYS_287"	}, /* 287 */
+	{ 5,	0,	printargs,		"SYS_288"	}, /* 288 */
+	{ 5,	0,	printargs,		"SYS_289"	}, /* 289 */
+	{ 5,	0,	printargs,		"SYS_290"	}, /* 290 */
+	{ 5,	0,	printargs,		"SYS_291"	}, /* 291 */
+	{ 5,	0,	printargs,		"SYS_292"	}, /* 292 */
+	{ 5,	0,	printargs,		"SYS_293"	}, /* 293 */
+	{ 5,	0,	printargs,		"SYS_294"	}, /* 294 */
+	{ 5,	0,	printargs,		"SYS_295"	}, /* 295 */
+	{ 5,	0,	printargs,		"SYS_296"	}, /* 296 */
+	{ 5,	0,	printargs,		"SYS_297"	}, /* 297 */
 	{ 5,	0,	printargs,		"SYS_298"	}, /* 298 */
 	{ 5,	0,	printargs,		"SYS_299"	}, /* 299 */
+	{ 5,	0,	printargs,		"SYS_300"	}, /* 300 */
+	{ 5,	0,	printargs,		"SYS_301"	}, /* 301 */
+	{ 5,	0,	printargs,		"SYS_302"	}, /* 302 */
+	{ 5,	0,	printargs,		"SYS_303"	}, /* 303 */
+	{ 5,	0,	printargs,		"SYS_304"	}, /* 304 */
+	{ 5,	0,	printargs,		"SYS_305"	}, /* 305 */
+	{ 5,	0,	printargs,		"SYS_306"	}, /* 306 */
+	{ 5,	0,	printargs,		"SYS_307"	}, /* 307 */
+	{ 5,	0,	printargs,		"SYS_308"	}, /* 308 */
+	{ 5,	0,	printargs,		"SYS_309"	}, /* 309 */
+	{ 5,	0,	printargs,		"SYS_310"	}, /* 310 */
+	{ 5,	0,	printargs,		"SYS_311"	}, /* 311 */
+	{ 5,	0,	printargs,		"SYS_312"	}, /* 312 */
+	{ 5,	0,	printargs,		"SYS_313"	}, /* 313 */
+	{ 5,	0,	printargs,		"SYS_314"	}, /* 314 */
+	{ 5,	0,	printargs,		"SYS_315"	}, /* 315 */
+	{ 5,	0,	printargs,		"SYS_316"	}, /* 316 */
+	{ 5,	0,	printargs,		"SYS_317"	}, /* 317 */
+	{ 5,	0,	printargs,		"SYS_318"	}, /* 318 */
+	{ 5,	0,	printargs,		"SYS_319"	}, /* 319 */
+	{ 5,	0,	printargs,		"SYS_320"	}, /* 320 */
+	{ 5,	0,	printargs,		"SYS_321"	}, /* 321 */
+	{ 5,	0,	printargs,		"SYS_322"	}, /* 322 */
+	{ 5,	0,	printargs,		"SYS_323"	}, /* 323 */
+	{ 5,	0,	printargs,		"SYS_324"	}, /* 324 */
+	{ 5,	0,	printargs,		"SYS_325"	}, /* 325 */
+	{ 5,	0,	printargs,		"SYS_326"	}, /* 326 */
+	{ 5,	0,	printargs,		"SYS_327"	}, /* 327 */
+	{ 5,	0,	printargs,		"SYS_328"	}, /* 328 */
+	{ 5,	0,	printargs,		"SYS_329"	}, /* 329 */
+	{ 5,	0,	printargs,		"SYS_330"	}, /* 330 */
+	{ 5,	0,	printargs,		"SYS_331"	}, /* 331 */
+	{ 5,	0,	printargs,		"SYS_332"	}, /* 332 */
+	{ 5,	0,	printargs,		"SYS_333"	}, /* 333 */
+	{ 5,	0,	printargs,		"SYS_334"	}, /* 334 */
+	{ 5,	0,	printargs,		"SYS_335"	}, /* 335 */
+	{ 5,	0,	printargs,		"SYS_336"	}, /* 336 */
+	{ 5,	0,	printargs,		"SYS_337"	}, /* 337 */
+	{ 5,	0,	printargs,		"SYS_338"	}, /* 338 */
+	{ 5,	0,	printargs,		"SYS_339"	}, /* 339 */
+	{ 5,	0,	printargs,		"SYS_340"	}, /* 340 */
+	{ 5,	0,	printargs,		"SYS_341"	}, /* 341 */
+	{ 5,	0,	printargs,		"SYS_342"	}, /* 342 */
+	{ 5,	0,	printargs,		"SYS_343"	}, /* 343 */
+	{ 5,	0,	printargs,		"SYS_344"	}, /* 344 */
+	{ 5,	0,	printargs,		"SYS_345"	}, /* 345 */
+	{ 5,	0,	printargs,		"SYS_346"	}, /* 346 */
+	{ 5,	0,	printargs,		"SYS_347"	}, /* 347 */
+	{ 5,	0,	printargs,		"SYS_348"	}, /* 348 */
+	{ 5,	0,	printargs,		"SYS_349"	}, /* 349 */
+	{ 5,	0,	printargs,		"SYS_350"	}, /* 350 */
+	{ 5,	0,	printargs,		"SYS_351"	}, /* 351 */
+	{ 5,	0,	printargs,		"SYS_352"	}, /* 352 */
+	{ 5,	0,	printargs,		"SYS_353"	}, /* 353 */
+	{ 5,	0,	printargs,		"SYS_354"	}, /* 354 */
+	{ 5,	0,	printargs,		"SYS_355"	}, /* 355 */
+	{ 5,	0,	printargs,		"SYS_356"	}, /* 356 */
+	{ 5,	0,	printargs,		"SYS_357"	}, /* 357 */
+	{ 5,	0,	printargs,		"SYS_358"	}, /* 358 */
+	{ 5,	0,	printargs,		"SYS_359"	}, /* 359 */
+	{ 5,	0,	printargs,		"SYS_360"	}, /* 360 */
+	{ 5,	0,	printargs,		"SYS_361"	}, /* 361 */
+	{ 5,	0,	printargs,		"SYS_362"	}, /* 362 */
+	{ 5,	0,	printargs,		"SYS_363"	}, /* 363 */
+	{ 5,	0,	printargs,		"SYS_364"	}, /* 364 */
+	{ 5,	0,	printargs,		"SYS_365"	}, /* 365 */
+	{ 5,	0,	printargs,		"SYS_366"	}, /* 366 */
+	{ 5,	0,	printargs,		"SYS_367"	}, /* 367 */
+	{ 5,	0,	printargs,		"SYS_368"	}, /* 368 */
+	{ 5,	0,	printargs,		"SYS_369"	}, /* 369 */
+	{ 5,	0,	printargs,		"SYS_370"	}, /* 370 */
+	{ 5,	0,	printargs,		"SYS_371"	}, /* 371 */
+	{ 5,	0,	printargs,		"SYS_372"	}, /* 372 */
+	{ 5,	0,	printargs,		"SYS_373"	}, /* 373 */
+	{ 5,	0,	printargs,		"SYS_374"	}, /* 374 */
+	{ 5,	0,	printargs,		"SYS_375"	}, /* 375 */
+	{ 5,	0,	printargs,		"SYS_376"	}, /* 376 */
+	{ 5,	0,	printargs,		"SYS_377"	}, /* 377 */
+	{ 5,	0,	printargs,		"SYS_378"	}, /* 378 */
+	{ 5,	0,	printargs,		"SYS_379"	}, /* 379 */
+	{ 5,	0,	printargs,		"SYS_380"	}, /* 380 */
+	{ 5,	0,	printargs,		"SYS_381"	}, /* 381 */
+	{ 5,	0,	printargs,		"SYS_382"	}, /* 382 */
+	{ 5,	0,	printargs,		"SYS_383"	}, /* 383 */
+	{ 5,	0,	printargs,		"SYS_384"	}, /* 384 */
+	{ 5,	0,	printargs,		"SYS_385"	}, /* 385 */
+	{ 5,	0,	printargs,		"SYS_386"	}, /* 386 */
+	{ 5,	0,	printargs,		"SYS_387"	}, /* 387 */
+	{ 5,	0,	printargs,		"SYS_388"	}, /* 388 */
+	{ 5,	0,	printargs,		"SYS_389"	}, /* 389 */
+	{ 5,	0,	printargs,		"SYS_390"	}, /* 390 */
+	{ 5,	0,	printargs,		"SYS_391"	}, /* 391 */
+	{ 5,	0,	printargs,		"SYS_392"	}, /* 392 */
+	{ 5,	0,	printargs,		"SYS_393"	}, /* 393 */
+	{ 5,	0,	printargs,		"SYS_394"	}, /* 394 */
+	{ 5,	0,	printargs,		"SYS_395"	}, /* 395 */
+	{ 5,	0,	printargs,		"SYS_396"	}, /* 396 */
+	{ 5,	0,	printargs,		"SYS_397"	}, /* 397 */
+	{ 5,	0,	printargs,		"SYS_398"	}, /* 398 */
+	{ 5,	0,	printargs,		"SYS_399"	}, /* 399 */
+
+#if SYS_ipc_subcall != 400
+ #error fix me
+#endif
+	{ 8,	0,	printargs,		"socket_subcall"}, /* 400 */
+	{ 3,	TN,	sys_socket,		"socket"	}, /* 401 */
+	{ 3,	TN,	sys_bind,		"bind"		}, /* 402 */
+	{ 3,	TN,	sys_connect,		"connect"	}, /* 403 */
+	{ 2,	TN,	sys_listen,		"listen"	}, /* 404 */
+	{ 3,	TN,	sys_accept,		"accept"	}, /* 405 */
+	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 406 */
+	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 407 */
+	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 408 */
+	{ 4,	TN,	sys_send,		"send"		}, /* 409 */
+	{ 4,	TN,	sys_recv,		"recv"		}, /* 410 */
+	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 411 */
+	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 412 */
+	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 413 */
+	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 414 */
+	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 415 */
+	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 416 */
+	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 417 */
+
+#if SYS_ipc_subcall != 418
+ #error fix me
+#endif
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 418 */
Index: strace/linux/sh64/syscallent.h
diff -u strace/linux/sh64/syscallent.h:1.5 strace/linux/sh64/syscallent.h:1.6
--- strace/linux/sh64/syscallent.h:1.5	Wed Aug  3 04:27:34 2005
+++ strace/linux/sh64/syscallent.h	Mon Aug 21 23:17:33 2006
@@ -25,7 +25,7 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *
- *	$Id: syscallent.h,v 1.5 2005/08/03 11:27:34 roland Exp $
+ *	$Id: syscallent.h,v 1.6 2006/08/22 06:17:33 roland Exp $
  */
 
 	{ 0,	0,	sys_setup,		"setup"		}, /* 0 */
@@ -328,22 +328,130 @@
 	{ 5,	0,	printargs,		"SYS_297"	}, /* 297 */
 	{ 5,	0,	printargs,		"SYS_298"	}, /* 298 */
 	{ 5,	0,	printargs,		"SYS_299"	}, /* 299 */
-	{ 8,	0,	printargs,		"socket_subcall"}, /* 300 */
-	{ 3,	TN,	sys_socket,		"socket"	}, /* 301 */
-	{ 3,	TN,	sys_bind,		"bind"		}, /* 302 */
-	{ 3,	TN,	sys_connect,		"connect"	}, /* 303 */
-	{ 2,	TN,	sys_listen,		"listen"	}, /* 304 */
-	{ 3,	TN,	sys_accept,		"accept"	}, /* 305 */
-	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 306 */
-	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 307 */
-	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 308 */
-	{ 4,	TN,	sys_send,		"send"		}, /* 309 */
-	{ 4,	TN,	sys_recv,		"recv"		}, /* 310 */
-	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 311 */
-	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 312 */
-	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 313 */
-	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 314 */
-	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 315 */
-	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 316 */
-	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 317 */
-	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 318 */
+	{ 5,	0,	printargs,		"SYS_300"	}, /* 300 */
+	{ 5,	0,	printargs,		"SYS_301"	}, /* 301 */
+	{ 5,	0,	printargs,		"SYS_302"	}, /* 302 */
+	{ 5,	0,	printargs,		"SYS_303"	}, /* 303 */
+	{ 5,	0,	printargs,		"SYS_304"	}, /* 304 */
+	{ 5,	0,	printargs,		"SYS_305"	}, /* 305 */
+	{ 5,	0,	printargs,		"SYS_306"	}, /* 306 */
+	{ 5,	0,	printargs,		"SYS_307"	}, /* 307 */
+	{ 5,	0,	printargs,		"SYS_308"	}, /* 308 */
+	{ 5,	0,	printargs,		"SYS_309"	}, /* 309 */
+	{ 5,	0,	printargs,		"SYS_310"	}, /* 310 */
+	{ 5,	0,	printargs,		"SYS_311"	}, /* 311 */
+	{ 5,	0,	printargs,		"SYS_312"	}, /* 312 */
+	{ 5,	0,	printargs,		"SYS_313"	}, /* 313 */
+	{ 5,	0,	printargs,		"SYS_314"	}, /* 314 */
+	{ 5,	0,	printargs,		"SYS_315"	}, /* 315 */
+	{ 5,	0,	printargs,		"SYS_316"	}, /* 316 */
+	{ 5,	0,	printargs,		"SYS_317"	}, /* 317 */
+	{ 5,	0,	printargs,		"SYS_318"	}, /* 318 */
+	{ 5,	0,	printargs,		"SYS_319"	}, /* 319 */
+	{ 5,	0,	printargs,		"SYS_320"	}, /* 320 */
+	{ 5,	0,	printargs,		"SYS_321"	}, /* 321 */
+	{ 5,	0,	printargs,		"SYS_322"	}, /* 322 */
+	{ 5,	0,	printargs,		"SYS_323"	}, /* 323 */
+	{ 5,	0,	printargs,		"SYS_324"	}, /* 324 */
+	{ 5,	0,	printargs,		"SYS_325"	}, /* 325 */
+	{ 5,	0,	printargs,		"SYS_326"	}, /* 326 */
+	{ 5,	0,	printargs,		"SYS_327"	}, /* 327 */
+	{ 5,	0,	printargs,		"SYS_328"	}, /* 328 */
+	{ 5,	0,	printargs,		"SYS_329"	}, /* 329 */
+	{ 5,	0,	printargs,		"SYS_330"	}, /* 330 */
+	{ 5,	0,	printargs,		"SYS_331"	}, /* 331 */
+	{ 5,	0,	printargs,		"SYS_332"	}, /* 332 */
+	{ 5,	0,	printargs,		"SYS_333"	}, /* 333 */
+	{ 5,	0,	printargs,		"SYS_334"	}, /* 334 */
+	{ 5,	0,	printargs,		"SYS_335"	}, /* 335 */
+	{ 5,	0,	printargs,		"SYS_336"	}, /* 336 */
+	{ 5,	0,	printargs,		"SYS_337"	}, /* 337 */
+	{ 5,	0,	printargs,		"SYS_338"	}, /* 338 */
+	{ 5,	0,	printargs,		"SYS_339"	}, /* 339 */
+	{ 5,	0,	printargs,		"SYS_340"	}, /* 340 */
+	{ 5,	0,	printargs,		"SYS_341"	}, /* 341 */
+	{ 5,	0,	printargs,		"SYS_342"	}, /* 342 */
+	{ 5,	0,	printargs,		"SYS_343"	}, /* 343 */
+	{ 5,	0,	printargs,		"SYS_344"	}, /* 344 */
+	{ 5,	0,	printargs,		"SYS_345"	}, /* 345 */
+	{ 5,	0,	printargs,		"SYS_346"	}, /* 346 */
+	{ 5,	0,	printargs,		"SYS_347"	}, /* 347 */
+	{ 5,	0,	printargs,		"SYS_348"	}, /* 348 */
+	{ 5,	0,	printargs,		"SYS_349"	}, /* 349 */
+	{ 5,	0,	printargs,		"SYS_350"	}, /* 350 */
+	{ 5,	0,	printargs,		"SYS_351"	}, /* 351 */
+	{ 5,	0,	printargs,		"SYS_352"	}, /* 352 */
+	{ 5,	0,	printargs,		"SYS_353"	}, /* 353 */
+	{ 5,	0,	printargs,		"SYS_354"	}, /* 354 */
+	{ 5,	0,	printargs,		"SYS_355"	}, /* 355 */
+	{ 5,	0,	printargs,		"SYS_356"	}, /* 356 */
+	{ 5,	0,	printargs,		"SYS_357"	}, /* 357 */
+	{ 5,	0,	printargs,		"SYS_358"	}, /* 358 */
+	{ 5,	0,	printargs,		"SYS_359"	}, /* 359 */
+	{ 5,	0,	printargs,		"SYS_360"	}, /* 360 */
+	{ 5,	0,	printargs,		"SYS_361"	}, /* 361 */
+	{ 5,	0,	printargs,		"SYS_362"	}, /* 362 */
+	{ 5,	0,	printargs,		"SYS_363"	}, /* 363 */
+	{ 5,	0,	printargs,		"SYS_364"	}, /* 364 */
+	{ 5,	0,	printargs,		"SYS_365"	}, /* 365 */
+	{ 5,	0,	printargs,		"SYS_366"	}, /* 366 */
+	{ 5,	0,	printargs,		"SYS_367"	}, /* 367 */
+	{ 5,	0,	printargs,		"SYS_368"	}, /* 368 */
+	{ 5,	0,	printargs,		"SYS_369"	}, /* 369 */
+	{ 5,	0,	printargs,		"SYS_370"	}, /* 370 */
+	{ 5,	0,	printargs,		"SYS_371"	}, /* 371 */
+	{ 5,	0,	printargs,		"SYS_372"	}, /* 372 */
+	{ 5,	0,	printargs,		"SYS_373"	}, /* 373 */
+	{ 5,	0,	printargs,		"SYS_374"	}, /* 374 */
+	{ 5,	0,	printargs,		"SYS_375"	}, /* 375 */
+	{ 5,	0,	printargs,		"SYS_376"	}, /* 376 */
+	{ 5,	0,	printargs,		"SYS_377"	}, /* 377 */
+	{ 5,	0,	printargs,		"SYS_378"	}, /* 378 */
+	{ 5,	0,	printargs,		"SYS_379"	}, /* 379 */
+	{ 5,	0,	printargs,		"SYS_380"	}, /* 380 */
+	{ 5,	0,	printargs,		"SYS_381"	}, /* 381 */
+	{ 5,	0,	printargs,		"SYS_382"	}, /* 382 */
+	{ 5,	0,	printargs,		"SYS_383"	}, /* 383 */
+	{ 5,	0,	printargs,		"SYS_384"	}, /* 384 */
+	{ 5,	0,	printargs,		"SYS_385"	}, /* 385 */
+	{ 5,	0,	printargs,		"SYS_386"	}, /* 386 */
+	{ 5,	0,	printargs,		"SYS_387"	}, /* 387 */
+	{ 5,	0,	printargs,		"SYS_388"	}, /* 388 */
+	{ 5,	0,	printargs,		"SYS_389"	}, /* 389 */
+	{ 5,	0,	printargs,		"SYS_390"	}, /* 390 */
+	{ 5,	0,	printargs,		"SYS_391"	}, /* 391 */
+	{ 5,	0,	printargs,		"SYS_392"	}, /* 392 */
+	{ 5,	0,	printargs,		"SYS_393"	}, /* 393 */
+	{ 5,	0,	printargs,		"SYS_394"	}, /* 394 */
+	{ 5,	0,	printargs,		"SYS_395"	}, /* 395 */
+	{ 5,	0,	printargs,		"SYS_396"	}, /* 396 */
+	{ 5,	0,	printargs,		"SYS_397"	}, /* 397 */
+	{ 5,	0,	printargs,		"SYS_398"	}, /* 398 */
+	{ 5,	0,	printargs,		"SYS_399"	}, /* 399 */
+
+#if SYS_ipc_subcall != 400
+ #error fix me
+#endif
+	{ 8,	0,	printargs,		"socket_subcall"}, /* 400 */
+	{ 3,	TN,	sys_socket,		"socket"	}, /* 401 */
+	{ 3,	TN,	sys_bind,		"bind"		}, /* 402 */
+	{ 3,	TN,	sys_connect,		"connect"	}, /* 403 */
+	{ 2,	TN,	sys_listen,		"listen"	}, /* 404 */
+	{ 3,	TN,	sys_accept,		"accept"	}, /* 405 */
+	{ 3,	TN,	sys_getsockname,	"getsockname"	}, /* 406 */
+	{ 3,	TN,	sys_getpeername,	"getpeername"	}, /* 407 */
+	{ 4,	TN,	sys_socketpair,		"socketpair"	}, /* 408 */
+	{ 4,	TN,	sys_send,		"send"		}, /* 409 */
+	{ 4,	TN,	sys_recv,		"recv"		}, /* 410 */
+	{ 6,	TN,	sys_sendto,		"sendto"	}, /* 411 */
+	{ 6,	TN,	sys_recvfrom,		"recvfrom"	}, /* 412 */
+	{ 2,	TN,	sys_shutdown,		"shutdown"	}, /* 413 */
+	{ 5,	TN,	sys_setsockopt,		"setsockopt"	}, /* 414 */
+	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 415 */
+	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 416 */
+	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 417 */
+
+#if SYS_ipc_subcall != 418
+ #error fix me
+#endif
+	{ 4,	0,	printargs,		"ipc_subcall"	}, /* 418 */
--- strace/linux/ia64/syscallent.h.jj	2006-01-12 17:34:50.000000000 -0500
+++ strace/linux/ia64/syscallent.h	2006-11-20 07:30:58.000000000 -0500
@@ -246,106 +246,10 @@
    starts where that one leaves off.
 */
 
-	{ 8,	0,	printargs,		"SYS_343"	}, /* 343 */
-	{ 8,	0,	printargs,		"SYS_344"	}, /* 344 */
-	{ 8,	0,	printargs,		"SYS_345"	}, /* 345 */
-	{ 8,	0,	printargs,		"SYS_346"	}, /* 346 */
-	{ 8,	0,	printargs,		"SYS_347"	}, /* 347 */
-	{ 8,	0,	printargs,		"SYS_348"	}, /* 348 */
-	{ 8,	0,	printargs,		"SYS_349"	}, /* 349 */
-	{ 8,	0,	printargs,		"SYS_350"	}, /* 350 */
-	{ 8,	0,	printargs,		"SYS_351"	}, /* 351 */
-	{ 8,	0,	printargs,		"SYS_352"	}, /* 352 */
-	{ 8,	0,	printargs,		"SYS_353"	}, /* 353 */
-	{ 8,	0,	printargs,		"SYS_354"	}, /* 354 */
-	{ 8,	0,	printargs,		"SYS_355"	}, /* 355 */
-	{ 8,	0,	printargs,		"SYS_356"	}, /* 356 */
-	{ 8,	0,	printargs,		"SYS_357"	}, /* 357 */
-	{ 8,	0,	printargs,		"SYS_358"	}, /* 358 */
-	{ 8,	0,	printargs,		"SYS_359"	}, /* 359 */
-	{ 8,	0,	printargs,		"SYS_360"	}, /* 360 */
-	{ 8,	0,	printargs,		"SYS_361"	}, /* 361 */
-	{ 8,	0,	printargs,		"SYS_362"	}, /* 362 */
-	{ 8,	0,	printargs,		"SYS_363"	}, /* 363 */
-	{ 8,	0,	printargs,		"SYS_364"	}, /* 364 */
-	{ 8,	0,	printargs,		"SYS_365"	}, /* 365 */
-	{ 8,	0,	printargs,		"SYS_366"	}, /* 366 */
-	{ 8,	0,	printargs,		"SYS_367"	}, /* 367 */
-	{ 8,	0,	printargs,		"SYS_368"	}, /* 368 */
-	{ 8,	0,	printargs,		"SYS_369"	}, /* 369 */
-	{ 8,	0,	printargs,		"SYS_370"	}, /* 370 */
-	{ 8,	0,	printargs,		"SYS_371"	}, /* 371 */
-	{ 8,	0,	printargs,		"SYS_372"	}, /* 372 */
-	{ 8,	0,	printargs,		"SYS_373"	}, /* 373 */
-	{ 8,	0,	printargs,		"SYS_374"	}, /* 374 */
-	{ 8,	0,	printargs,		"SYS_375"	}, /* 375 */
-	{ 8,	0,	printargs,		"SYS_376"	}, /* 376 */
-	{ 8,	0,	printargs,		"SYS_377"	}, /* 377 */
-	{ 8,	0,	printargs,		"SYS_378"	}, /* 378 */
-	{ 8,	0,	printargs,		"SYS_379"	}, /* 379 */
-	{ 8,	0,	printargs,		"SYS_380"	}, /* 380 */
-	{ 8,	0,	printargs,		"SYS_381"	}, /* 381 */
-	{ 8,	0,	printargs,		"SYS_382"	}, /* 382 */
-	{ 8,	0,	printargs,		"SYS_383"	}, /* 383 */
-	{ 8,	0,	printargs,		"SYS_384"	}, /* 384 */
-	{ 8,	0,	printargs,		"SYS_385"	}, /* 385 */
-	{ 8,	0,	printargs,		"SYS_386"	}, /* 386 */
-	{ 8,	0,	printargs,		"SYS_387"	}, /* 387 */
-	{ 8,	0,	printargs,		"SYS_388"	}, /* 388 */
-	{ 8,	0,	printargs,		"SYS_389"	}, /* 389 */
-	{ 8,	0,	printargs,		"SYS_390"	}, /* 390 */
-	{ 8,	0,	printargs,		"SYS_391"	}, /* 391 */
-	{ 8,	0,	printargs,		"SYS_392"	}, /* 392 */
-	{ 8,	0,	printargs,		"SYS_393"	}, /* 393 */
-	{ 8,	0,	printargs,		"SYS_394"	}, /* 394 */
-	{ 8,	0,	printargs,		"SYS_395"	}, /* 395 */
-	{ 8,	0,	printargs,		"SYS_396"	}, /* 396 */
-	{ 8,	0,	printargs,		"SYS_397"	}, /* 397 */
-	{ 8,	0,	printargs,		"SYS_398"	}, /* 398 */
-	{ 8,	0,	printargs,		"SYS_399"	}, /* 399 */
-	{ 8,	0,	printargs,		"SYS_400"	}, /* 400 */
-	{ 8,	0,	printargs,		"SYS_401"	}, /* 401 */
-	{ 8,	0,	printargs,		"SYS_402"	}, /* 402 */
-	{ 8,	0,	printargs,		"SYS_403"	}, /* 403 */
-	{ 8,	0,	printargs,		"SYS_404"	}, /* 404 */
-	{ 8,	0,	printargs,		"SYS_405"	}, /* 405 */
-	{ 8,	0,	printargs,		"SYS_406"	}, /* 406 */
-	{ 8,	0,	printargs,		"SYS_407"	}, /* 407 */
-	{ 8,	0,	printargs,		"SYS_408"	}, /* 408 */
-	{ 8,	0,	printargs,		"SYS_409"	}, /* 409 */
-	{ 8,	0,	printargs,		"SYS_410"	}, /* 410 */
-	{ 8,	0,	printargs,		"SYS_411"	}, /* 411 */
-	{ 8,	0,	printargs,		"SYS_412"	}, /* 412 */
-	{ 8,	0,	printargs,		"SYS_413"	}, /* 413 */
-	{ 8,	0,	printargs,		"SYS_414"	}, /* 414 */
-	{ 8,	0,	printargs,		"SYS_415"	}, /* 415 */
-	{ 8,	0,	printargs,		"SYS_416"	}, /* 416 */
-	{ 8,	0,	printargs,		"SYS_417"	}, /* 417 */
-	{ 8,	0,	printargs,		"SYS_418"	}, /* 418 */
-	{ 8,	0,	printargs,		"SYS_419"	}, /* 419 */
-	{ 8,	0,	printargs,		"SYS_420"	}, /* 420 */
-	{ 8,	0,	printargs,		"SYS_421"	}, /* 421 */
-	{ 8,	0,	printargs,		"SYS_422"	}, /* 422 */
-	{ 8,	0,	printargs,		"SYS_423"	}, /* 423 */
-	{ 8,	0,	printargs,		"SYS_424"	}, /* 424 */
-	{ 8,	0,	printargs,		"SYS_425"	}, /* 425 */
-	{ 8,	0,	printargs,		"SYS_426"	}, /* 426 */
-	{ 8,	0,	printargs,		"SYS_427"	}, /* 427 */
-	{ 8,	0,	printargs,		"SYS_428"	}, /* 428 */
-	{ 8,	0,	printargs,		"SYS_429"	}, /* 429 */
-	{ 8,	0,	printargs,		"SYS_430"	}, /* 430 */
-	{ 8,	0,	printargs,		"SYS_431"	}, /* 431 */
-	{ 8,	0,	printargs,		"SYS_432"	}, /* 432 */
-	{ 8,	0,	printargs,		"SYS_433"	}, /* 433 */
-	{ 8,	0,	printargs,		"SYS_434"	}, /* 434 */
-	{ 8,	0,	printargs,		"SYS_435"	}, /* 435 */
-	{ 8,	0,	printargs,		"SYS_436"	}, /* 436 */
-	{ 8,	0,	printargs,		"SYS_437"	}, /* 437 */
-	{ 8,	0,	printargs,		"SYS_438"	}, /* 438 */
-	{ 8,	0,	printargs,		"SYS_439"	}, /* 439 */
-	{ 8,	0,	printargs,		"SYS_440"	}, /* 440 */
-	{ 8,	0,	printargs,		"SYS_441"	}, /* 441 */
-	{ 8,	0,	printargs,		"SYS_442"	}, /* 442 */
+#if SYS_ipc_subcall + SYS_ipc_nsubcalls != 443
+# error fix me
+#endif
+
 	{ 8,	0,	printargs,		"SYS_443"	}, /* 443 */
 	{ 8,	0,	printargs,		"SYS_444"	}, /* 444 */
 	{ 8,	0,	printargs,		"SYS_445"	}, /* 445 */
